Services/Mail/classes/AutoResponder/class.ilAutoResponderDatabaseRepository.php

```diff
@@ -0,0 +1,115 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ ********************************************************************
+ */
+class ilAutoResponderDatabaseRepository implements ilAutoResponderRepository
+{
+    public const TABLE_NAME = 'auto_responder';
+
+    protected ilDBInterface $db;
+
+    public function __construct()
+    {
+        global $DIC;
```

Comment:
This class should not rely on `global $DIC`. Please pass an `ilDBInterface` as a constructor dependency. Of course we have to decide how to create the repository with the database adapter.

We should create a global `mail()` service and make it accessible in the `DIC` (`\ILIAS\DI\Container`) similar to the other components. Maybe we could also get rid of `\ilInitialisation::initMail.